aioamqp/channel.py

```diff
@@ -109,7 +123,12 @@ def open(self, timeout=None):
         request = amqp_frame.AmqpEncoder()
         request.write_shortstr('')
         fut = self._set_waiter('open')
-        yield from self._write_frame(frame, request, no_wait=False, timeout=timeout, no_check_open=True)
```

Comment:
Another alternative: _write_frame is decorated as a coroutine, but it does no yield from. Therefore it executes until it returns or an exception is raised. No one else has the opportunity to look up the waiter future created just before the _write_frame call and reference it, so the future could just as well be created after the _write_frame call so it wouldn't need any exception handling.

The @coroutine could be dropped from _write_frame in the same go as it's class internal.